### ISSUE TITLE
Map.ofEntries equivalent for fastutil maps

### ISSUE THREAD

**Issue by viciscat** [vigna/fastutil#329]
Adding something like [Map#ofEntries](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/Map.html#ofEntries(java.util.Map.Entry...)) to the fastutil maps would be neat in my humble opinion.
Right now you gotta do ugly stuff like `Int2ObjectMaps.unmodifiable(new Int2ObjectOpenHashMap(Map.ofEntries(...)))`
I could make a PR if you don't have time to.
(btw great library much love ❤️)

**Comment by vigna** [comment]
Good point, a PR would be welcome! I have the same problem.

**Comment by viciscat** [comment]
sweet! Should the methods be added in the `Map` interfaces or `Maps` (with an s) classes

**Comment by vigna** [comment]
Good point. I'd follow the JCF and put in in Map. But what concrete class do you intend to return, an Open2OpenHashMap in an unmodifiable wrapper?

**Comment by viciscat** [comment]
Not sure honestly... I'm not the data structure champion. Maybe an array implementation? I heard that it's good when you just access and don't write in it. Did not do a benchmark tho. The implementation of `Map#ofEntries` is just an array that alternates between key and value. Won't work for most maps (unless it's the same primitive type for the key and value)

**Comment by vigna** [comment]
Well, if it's immutable and holds at most 10 entries I think an ArrayMap is perfectly fine. It could even be the case of creating an ImmutableArrayMap, but that would mean another 90 classes. Maybe as an internal class of Map, just copying the necessary code from ArrayMap?

**Comment by viciscat** [comment]
What if it's more than 10 entries? HashMap?

**Comment by viciscat** [comment]
Or I could just return an ArrayMap / OpenHashMap in an unmodifiable wrapper depending on the number of entries. That would be the easiest to write.

**Comment by vigna** [comment]
Oh you're right. Map.of has at most 10 elements, but Map.ofEntries has variable length.

Yes, you could to ArrayMap up to, say, 8 elements, and, an OpenHashMap otherwise.

**Comment by viciscat** [comment]
I'm guessing tests would be wanted? I don't see any test classes for the Map interfaces. Do I create a new one? Do I do it like `Primitive2PrimitiveMapTest` and `Primitive2ObjectMapTest`?

**Comment by vigna** [comment]
Yes, MapTest sounds right. Like Int2IntMapTest and Object2ObjectMapTest should be enough. 

Please double check that I didn't put in Maps stuff that in Java is in Map... Or we might have to rethink the position.

**Comment by viciscat** [comment]
Hey having a little trouble compiling a jar... Running `ant jar` tries to compile the drv files and I don't think that's normal 😅

**Comment by vigna** [comment]
No, it shouldn't. Did you `make -s sources` first?

**Comment by viciscat** [comment]
Did not include `-s` since it isn't specified in the readme but yes I did.

**Comment by vigna** [comment]
Oh yeah that's just "silent" to avoid the mess. Platform? OS?

**Comment by viciscat** [comment]
Hey it's me again, turns out I just can't read errors. It seems some java files are not getting generated correctly? For example `Byte2ByteLinkedOpenHashMap.java` gets generated with just `OpenHashMap.drv` inside of it. It might be more WSL shenanigans but asking nonetheless in case it's a known issue.
![image](https://github.com/user-attachments/assets/418ee3eb-b52e-42ae-86b5-3cd5579e1827)

**Comment by vigna** [comment]
That scripting part is for UN*X variants. That's a WSL issue, I'm sorry but I can't help you...

**Comment by viciscat** [comment]
I can't seem to find anywhere in `gensources.sh` or the `makefile` how it detects and replaces the `drv` file that just contains the name of another `drv` file. Kinda hard to debug when I don't know how it's done

**Comment by vigna** [comment]
It might be a problem with different treatment of soft links, as the .drv files of non-plain hash maps are just soft links to the real ones.

**Comment by viciscat** [comment]
Hey I made this little workaround and it works great it seems. Should I commit that as well or no?
(The $1 at the end of the file is replaced by $TARGET_DRV_FILE)
```bash

TARGET_DRV_FILE=$1

# A workaround for Windows and WSL. Linked files only contain the "target" file's name on windows.
# So we check the first line to see if it contains ".drv". If so replace the target drv file by that.
FIRST_LINE=$(head -n 1 $TARGET_DRV_FILE)
if [[ $FIRST_LINE == *".drv"* ]]; then
  TARGET_DRV_FILE="drv/$FIRST_LINE"
fi
```

**Comment by vigna** [comment]
The problem is checking that it works everywhere.
